File: Java/프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int sum = brown + yellow;
        for (int i = 3; i < sum; i++) {
            int min = i;
            int max = sum / min;
            
            if ((max - 2) * (min - 2) == yellow) {
                answer[0] = max;
                answer[1] = min;
                break;
            }
        }
        return answer;
    }
}

File: Python/프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = [0] * 2
    sum = brown + yellow
    for i in range(3, sum):
        min = i
        max = sum // min
        if (min - 2) * (max - 2) == yellow:
            answer[0] = max
            answer[1] = min
            break

    return answer